src/framework/theme/components/popover/popover.directive.ts

```diff
@@ -0,0 +1,306 @@
+/**
+ * @license
+ * Copyright Akveo. All Rights Reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ */
+
+import { ComponentRef, Directive, ElementRef, HostListener, Input, OnDestroy, OnInit, TemplateRef,
+  Type,
+} from '@angular/core';
+import { NbPlacement, NbPositioningHelper } from './positioning.helper';
+import { NbPopoverComponent } from './popover.component';
+import { NbThemeService } from '../../services/theme.service';
+import { takeWhile } from 'rxjs/operators/takeWhile';
+import { NbPosition, NbAdjustmentHelper, NbAdjustment } from './adjustment.helper';
+import { NbPopoverMode, NbTriggerHelper } from './trigger.helper';
+
+/**
+ * Popover can be one of the following types:
+ * template, component or plain js string.
+ * So NbPopoverContent provides types alias for this purposes.
+ * */
+export type NbPopoverContent = string | TemplateRef<any> | Type<any>;
+
+/**
+ * Powerful popover directive, which provides the best UX for your users.
+ *
+ * @example Popover can accept different content such as:
+ * TemplateRef
+ *
+ * ```
+ * <button [nbPopover]="templateRef"></button>
+ * <ng-template #templateRef>
+ *   <span>Hello, Popover!</span>
+ * </ng-template>
+ * ```
+ *
+ * @example Custom components
+ *
+ * ```
+ * <button [nbPopover]="NbCardComponent"></button>
+ * ```
+ *
+ * @example Primitive types
+ *
+ * ```
+ * <button [nbPopover]="'Hello, Popover!'"></button>
```

Comment:
`<button nbPopover="Hello, Popover!"></button>`
no need to use [] and additional '' in all of your examples

Reply:
fixed